src/features/programSample/samples.ts

```diff
@@ -62,8 +62,13 @@ export const downloadSampleIndex = () =>
 export const downloadModemFirmware = (modemFirmware: ModemFirmware) =>
     downloadFile(modemFirmware.file);
 
-export const downloadSample = (sample: Sample) =>
+export const downloadSample = async (sample: Sample) => {
+    if (!existsSync(DOWNLOAD_FOLDER)) {
+        await mkdir(DOWNLOAD_FOLDER);
```

Comment:
Why not just use `mkdirSync`?

Reply:
That would block the thread?

Reply:
Isn't the same true for `existsSync` above? Either we say that these FS operations are fast enough and it doesn't matter in this place, then we can make both sync or we should make both async.

(And generally we should move all of these low level operations out of the renderer thread because they are critical there anyhow, but that is another thing.)

Reply:
Yeah, there is no `exists` in the fs/promises, but I guess the optimal way would be to use `access` or something similar to avoid the locking.